doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+
+.. code-block:: python
+
+   @pipeBase.timeMethod
+   def run(self, maskedImage):
+       """!Compute and return statistics for a masked image
+       @param[in] maskedImage: masked image (an lsst::afw::MaskedImage)
+       @return a pipeBase Struct containing:
+       - mean: mean of image plane
+       - meanErr: uncertainty in mean
+       - stdDev: standard deviation of image plane
+       - stdDevErr: uncertainty in standard deviation
+       """
+       statObj = afwMath.makeStatistics(maskedImage, afwMath.MEANCLIP | afwMath.STDEVCLIP | afwMath.ERRORS,
+                                        self._statsControl)
+       mean, meanErr = statObj.getResult(afwMath.MEANCLIP)
+       stdDev, stdDevErr = statObj.getResult(afwMath.STDEVCLIP)
+       self.log.info("clipped mean=%0.2f; meanErr=%0.2f; stdDev=%0.2f; stdDevErr=%0.2f" %
+                     (mean, meanErr, stdDev, stdDevErr))
+       return pipeBase.Struct(
+           mean=mean,
+           meanErr=meanErr,
+           stdDev=stdDev,
+           stdDevErr=stdDevErr,
+       )
+
+.. _creating-a-task-debug-variables:
+
+Debug variables
+===============
+
+Debug variables are variables the user may set while running your task, to enable additional debug output.
+To have your task support debug variables, have it import ``lsstDebug`` and call ``lsstDebug.Info(__name__).varname`` to get the debug variable ``varname`` specific to your task.
+If you look for a variable the user has not specified, it will have a value of `False`.
+For example, to look for a debug variable named "display":
+
+.. code-block:: python
+
+   import lsstDebug
+   display = lsstDebug.Info(__name__).display
+   if display:
+      # ...
+      pass
+
+.. FIXME create link when lsstDebug documentation is ready.
+.. See \ref baseDebug for more information about debug variables, including how to specify them while running a command-line task.
+
+.. _creating-a-task-docs:
+
+Documentation
+=============
+
+To be written.
+
+.. _creating-a-task-subtasks:
+
+Subtasks
+========
+
+Each subtask is specified in the configuration as a field of type `lsst.pex.config.ConfigurableField` or (less commonly) `lsst.pex.config.RegistryField`.
+There are advantages to each:
+
+- `lsst.pex.config.ConfigurableField` advantages:
+
+  - It is easier for the user to override settings of the subtask; simply use dotted name notation:
+
+    .. code-block:: python
+
+       config.configurableSubtask.subtaskParam1 = ...
+
+    In contrast, to override configuration for a subtask specified as an `lsst.pex.config.RegistryField` you must either specify the name of the subtask to configure:
+
+    .. code-block:: python
+
+       config.registrySubtask[nameOfSelectedSubtask].subtaskParam1 = ...
+
+    Or use the `active` attribute to modify the configuration of the currently selected (active) subtask:
+
+    .. code-block:: python
+
+       config.registrySubtask.active.subtaskParam1 = ...
```

Comment:
I don't understand this.  Is this basically an alias?  If so, how do we know that all subtasks have the same parameters?  It seems like a dangerous/confusing piece of syntactic sugar.